app/controllers/buyers/applications_controller.rb

```diff
@@ -191,8 +191,12 @@ def find_provider
     @provider = current_account
   end
 
-  def find_application_plan
-    @application_plan = @provider.application_plans.find params[:cinstance][:plan_id]
+  def find_accessible_services
+    @services = current_user.accessible_services
+  end
```

Comment:
We should still have the `includes(:application_plans)` for the `index` case as we used to,  in order to avoid a N+1.

https://github.com/3scale/porta/blob/c1dc253ac89c9b2e72dc064260ea9ae740c096ca/app/controllers/buyers/applications_controller.rb#L28

Something like this:
```ruby
def find_accessible_services
    @services = current_user.accessible_services
    @services = @services.includes(:application_plans) if action_name == 'index'
end
```

Reply:
Right. Although I think it would be better to call it in `index`?

Reply:
Yes, makes sense 😄